SpotGamma.cpp

```diff
@@ -1,249 +1,284 @@
+			// sc.AddMessageToLog(StateMessage, 0);
+            RequestState = REQUEST_SENT;
+            // sc.AddMessageToLog(sc.HTTPResponse, 1); // HTTP response for debugging
+            HttpResponseContent = sc.HTTPResponse;
+        }
     }
 
-    // Tokenize
-    scstringLine.Tokenize(",", columns);
-
-    int index = 0;
-    float priceLevel = 0; // Column A
-    char *label;          // Column B
-    SCString colorString; // Column C
-    for (char *cellChars : columns) {
-      if (index == 0) {
-        priceLevel = std::stof(cellChars);
-      } else if (index == 1) {
-        label = cellChars;
-      } else if (index == 2) {
-        SCString cellValue(cellChars);
-        if (cellValue.GetLength() == 7) {
-          colorString += cellValue.GetSubString(6, 1).GetChars();
+
+    if (RequestState == REQUEST_SENT) {
+        // StateMessage.Format("sc.Index: %d, UpdateIntervalInMin: %d, RequestState: %d", sc.Index, UpdateIntervalInMin, RequestState);
+	    // sc.AddMessageToLog(StateMessage, 0);
+        if (sc.HTTPResponse != "") {
+            RequestState = REQUEST_RECEIVED;
+			sc.AddMessageToLog("HTTP Request RECEIVED, Updating Levels", 1);
+            // sc.AddMessageToLog(sc.HTTPResponse, 1); // HTTP response for debugging
+            HttpResponseContent = sc.HTTPResponse;
         } else {
-          colorString += "CCCCCC";
+        // Request not completed, nothing to do
+			// sc.AddMessageToLog("Hanging Request", 0);
+        return;
         }
-      }
-
-      index++;
     }
 
-    int r, g, b;
-    sscanf_s(colorString.GetChars(), "%02x%02x%02x", &r, &g, &b);
-    const int color = colorOverride != RGB(0, 0, 0) ? colorOverride : RGB(r, g, b);
+    // reset state for next run
+    RequestState = REQUEST_NOT_SENT;
+	// StateMessage.Format("sc.Index: %d, UpdateIntervalInMin: %d, RequestState: %d", sc.Index, UpdateIntervalInMin, RequestState);
+	// sc.AddMessageToLog(StateMessage, 0);
+    // sc.AddMessageToLog("RequestState Reset: REQUEST_NOT_SENT", 0);
 
-    if (!(priceLevel > 0)) {
-      return;
-    }
+    // Tokenize
+    // Initialize vector to store tokens from CSV
+    std::vector<char*> tokens;
+    // Create input stream from CSV content retrieved from HTTP Response
+    std::istringstream input(HttpResponseContent.GetChars());
+    int LineNumber = 1;
+
+    // We're going to:
+    // - loop line by line
+    // - convert each line to SCString for processing
+    // - remove " characters as first and last characters which cause an error
+    // - tokenize the line 
+    // - initialize the Tool and tool settings
+    // - then process each token extracted from the current line
+    for (std::string line; getline(input,line);) {
+        // msg.Format("%s", line.c_str());
+        // sc.AddMessageToLog(msg,1);
+
+        SCString scline = line.c_str();
+		// May be a Google Sheets specific issue, unsure
+        scline = scline.GetSubString(scline.GetLength() - 2, 1);
+        scline.Tokenize("\",\"", tokens);
+
```

Comment:
I don't think we have quotes in our payload.
You'll need to tweak this to work with the actual CSV result.